thinc/backends/numpy_ops.pyx

```diff
@@ -93,21 +93,17 @@ class NumpyOps(Ops):
         return blis.py.gemm(x, y, out=out, trans1=trans1, trans2=trans2, beta=0.)
 
     def relu(self, np.ndarray X, inplace=False):
-        cdef np.ndarray Y
-        cdef weight_t* data
-        cdef size = X.size
+        cdef np.ndarray Y = _inplace_or_copy(X, inplace)
+
         if X.dtype == "float32":
-            if inplace:
-                Y = X
-            else:
-                Y = X.copy()
-            data = <weight_t*>Y.data
-            for i in range(size):
-                if data[i] < 0:
-                    data[i] = 0.
-            return Y
+            cpu_relu(<float *>Y.data, <int>Y.size)
+        elif X.dtype == "float64":
+            cpu_relu(<double *>Y.data, <int>Y.size)
         else:
-            return super().relu(X, inplace)
+            return super().relu(X, inplace=inplace)
+
+        return Y
```

Comment:
It reads a little weird to me that for 2 out of 3 `if` clauses, the `return` is at the end of the function, and for the other it's immediate.

(the same comment about some of the other function rewrites below)

Reply:
Fixed.